crates/re_space_view/src/data_query_blueprint.rs

```diff
@@ -315,76 +315,41 @@ impl<'a> QueryExpressionEvaluator<'a> {
-            // This list will be updated below during `update_overrides_recursive` by calling `choose_default_visualizers`
-            // on the space view.
-            let available_visualizers = self
-                .visualizable_entities_for_visualizer_systems
-                .iter()
-                .filter_map(|(visualizer, ents)| {
-                    if ents.contains(entity_path) {
-                        Some(*visualizer)
-                    } else {
-                        None
-                    }
-                })
-                .collect();
-
-            Some(data_results.insert(DataResultNode {
-                data_result: DataResult {
-                    entity_path: entity_path.clone(),
-                    visualizers: available_visualizers,
-                    is_group: false,
-                    direct_included: any_match,
-                    property_overrides: None,
-                },
-                children: Default::default(),
-            }))
-        } else {
-            None
-        };
-
-        let maybe_self_iter = if let Some(self_leaf) = self_leaf {
-            itertools::Either::Left(std::iter::once(self_leaf))
-        } else {
-            itertools::Either::Right(std::iter::empty())
-        };
+        // TODO(#5067): For now, we always start by setting visualizers to the full list of available visualizers.
+        // This is currently important for evaluating auto-properties during the space-view `on_frame_start`, which
+        // is called before the property-overrider has a chance to update this list.
+        // This list will be updated below during `update_overrides_recursive` by calling `choose_default_visualizers`
+        // on the space view.
+        let visualizers: SmallVec<[_; 4]> = self
+            .visualizable_entities_for_visualizer_systems
+            .iter()
+            .filter_map(|(visualizer, ents)| ents.contains(entity_path).then_some(*visualizer))
+            .collect();
 
-        let children: SmallVec<_> = maybe_self_iter
-            .chain(tree.children.values().filter_map(|subtree| {
+        let children: SmallVec<[_; 4]> = tree
+            .children
+            .values()
+            .filter_map(|subtree| {
                 self.add_entity_tree_to_data_results_recursive(subtree, data_results)
-            }))
+            })
             .collect();
 
-        // If the only child is the self-leaf, then we don't need to create a group
-        if children.is_empty() || children.len() == 1 && self_leaf.is_some() {
-            self_leaf
-        } else {
-            // The 'individual' properties of a group are the group overrides
+        // Ignore empty nodes.
+        // Since we recurse downwards, this prunes any branches that don't have anything to contribute to the scene
+        // and aren't directly included.
+        let direct_included = self.entity_path_filter.is_exact_included(entity_path);
```

Comment:
this is wrong and fixed in follow-up PR: `direct_included` should still just use `is_included`. But the `if` check here indeed wants to work with `is_exact_included`